lib/variable/include/scipp/variable/element_array_model.h

```diff
@@ -31,6 +32,29 @@ bool equals_impl(const T1 &view1, const T2 &view2) {
     return std::equal(view1.begin(), view1.end(), view2.begin(), view2.end());
 }
 
+template <class T> bool equals_nan(const T &a, const T &b) { return a == b; }
+
+template <class T1, class T2>
+bool equals_nan_impl(const T1 &view1, const T2 &view2) {
+  // TODO Use optimizations in case of contigous views (instead of slower
+  // ElementArrayView iteration). Add multi threading?
+  if constexpr (is_span_v<typename T1::value_type>)
+    return std::equal(
+        view1.begin(), view1.end(), view2.begin(), view2.end(),
+        [](const auto &a, const auto &b) { return equals_nan_impl(a, b); });
+  else
+    return std::equal(
+        view1.begin(), view1.end(), view2.begin(), view2.end(),
+        [](const auto &x, const auto &y) {
+          if constexpr (std::is_floating_point_v<std::decay_t<decltype(x)>>) {
+            using numeric::isnan;
+            if (isnan(x) && isnan(y))
```

Comment:
Why is this needed? Does `equals_nan(float, float)` not contain this check?

Reply:
It just uses `operator==`, but you are right that it may be clearer to move this bit into that helper function.